From ddd84ca9143768f15adff987e406bb14ce21d850 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 18:59:46 -0400
Subject: [fs] btrfs: handle EAGAIN case properly in btrfs_drop_snapshot()

Message-id: <1407179109-13714-100-git-send-email-sandeen@redhat.com>
Patchwork-id: 87060
O-Subject: [RHEL7.1 PATCH 099/422] Btrfs: handle EAGAIN case properly in btrfs_drop_snapshot()
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Wang Shilong <wangsl.fnst@cn.fujitsu.com>

commit 90515e7f5d7d24cbb2a4038a3f1b5cfa2921aa17
Author: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
Date:   Tue Jan 7 17:26:58 2014 +0800

    Btrfs: handle EAGAIN case properly in btrfs_drop_snapshot()

    We may return early in btrfs_drop_snapshot(), we shouldn't
    call btrfs_std_err() for this case, fix it.

    Cc: stable@vger.kernel.org
    Signed-off-by: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/extent-tree.c b/fs/btrfs/extent-tree.c
index 41fe80b..77acc08 100644
--- a/fs/btrfs/extent-tree.c
+++ b/fs/btrfs/extent-tree.c
@@ -7835,7 +7835,7 @@ out:
   */
  if (!for_reloc && root_dropped == false)
   btrfs_add_dead_root(root);
- if (err)
+ if (err && err != -EAGAIN)
   btrfs_std_error(root->fs_info, err);
  return err;
 }
-- 
1.7.1